core/modules/ccontrol/MergingHandler.cc

```diff
@@ -198,6 +198,11 @@ bool MergingHandler::reset() {
     return true;
 }
 
+
+bool MergingHandler::scrubResults(int jobId, int attempt) {
+    return _infileMerger->scrubResults(jobId, attempt);
```

Comment:
I could keep a set of all jobAttempts that had been added to the table, and only try to delete rows for those jobAttempts that had been added.